yt/data_objects/tests/test_data_containers.py

```diff
@@ -0,0 +1,121 @@
+import shelve
+
+import numpy as np
+from nose.tools import assert_raises
+from numpy.testing import assert_array_equal
+
+from yt.data_objects.data_containers import YTDataContainer
+from yt.testing import assert_equal, fake_random_ds, fake_amr_ds, fake_particle_ds
+from yt.utilities.exceptions import YTFieldNotFound
+
+
+def test_yt_data_container():
+    # Test if ds could be None
+    with assert_raises(RuntimeError) as err:
+        YTDataContainer(None, None)
+    desired = 'Error: ds must be set either through class type or parameter' \
+              ' to the constructor'
+    assert_equal(str(err.exception), desired)
+
+    # Test if field_data key exists
+    ds = fake_random_ds(5)
+    proj = ds.proj("density", 0, data_source=ds.all_data())
+    assert_equal('px' in proj.keys(), True)
+    assert_equal('pz' in proj.keys(), False)
+
+    # Delete the key and check if exits
+    proj.__delitem__('px')
+    assert_equal('px' in proj.keys(), False)
+    proj.__delitem__('density')
+    assert_equal('density' in proj.keys(), False)
+
+    # Delete a non-existent field
+    with assert_raises(YTFieldNotFound) as ex:
+        proj.__delitem__('p_mass')
+    desired = "Could not find field '('stream', 'p_mass')' in UniformGridData."
+    assert_equal(str(ex.exception), desired)
+
+    # Test the convert method
+    assert_equal(proj.convert('HydroMethod'), -1)
+
+def test_write_out():
+    filename = "sphere.txt"
+    ds = fake_particle_ds()
+    sp = ds.sphere(ds.domain_center, 0.25)
+    sp.write_out(filename)
+
+    with open(filename, "r") as file:
+        file_row_1 = file.readline()
+        file_row_2 = file.readline()
+        file_row_2 = np.array(file_row_2.split('\t'), dtype=np.float64)
+    sorted_keys = sorted(sp.field_data.keys())
+    _keys = [str(k) for k in sorted_keys]
+    _keys = "\t".join(["#"] + _keys + ["\n"])
+    _data = [sp.field_data[k][0] for k in sorted_keys]
+
+    assert_equal(_keys, file_row_1)
+    assert_array_equal(_data, file_row_2)
+
+def test_save_object():
+    ds = fake_particle_ds()
+    sp = ds.sphere(ds.domain_center, 0.25)
+    sp.save_object("my_sphere_1", filename="test_save_obj")
+    obj = shelve.open("test_save_obj", protocol=-1)
+    loaded_sphere = obj["my_sphere_1"][1]
+    assert_array_equal(loaded_sphere.center, sp.center)
+    assert_equal(loaded_sphere.radius, sp.radius)
+    obj.close()
+
+    sp.save_object("my_sphere_2")
+
+def test_to_dataframe():
+    try:
+        import pandas as pd
```

Comment:
Does `import pandas as _` make flake8 happy?

Reply:
Rather than wrapping the test logic in a try/except, please use `yt.testing.requires_module`, which is a decorator you can apply to a test function to tell nose that the test function requires some optional module (in this case `pandas`).

You probably also need to install pandas on travis and appveyor to get this test to actually execute.